src/update-feeds/update-feeds.ts

```diff
@@ -186,5 +186,19 @@ export const processBatch = async (batch: ReadDapiWithIndexResponse[], providerN
     }
   });
 
-  return updateFeeds(getFeedsToUpdate(batch), chainId);
+  const feeds = getFeedsToUpdate(batch);
+
+  // CLear last update timestamps for feeds that don't need an update
+  for (const feed of feeds.filter((feed) => !feed.shouldUpdate)) {
```

Comment:
nit: I would filter the ones that don't need to be updated based on the missing data feed IDs from the feeds to be updated (e.g. something along these lines `batch.filter(dapi => !feeds.includes(dapi.dataFeedId))`).

But I don't mind your approach either.